lib/rules/jsx-no-invalid-rel.js

```diff
@@ -0,0 +1,190 @@
+    const start = match.index + valueRangeStart;
+    const end = start + match[0].length;
+    res.push({
+      reportingValue: `"${match[1]}"`,
+      value: match[1],
+      range: [start, end]
+    });
+  }
+
+  return res;
+}
+
+function checkLiteralValueNode(context, node, parentNodeName) {
+  if (typeof node.value !== 'string') {
+    return context.report({
+      node,
+      messageId: 'onlyStrings',
+      fix(fixer) {
+        return fixer.remove(node.parent.parent);
+      }
+    });
+  }
+
+  if (!node.value.trim()) {
+    return context.report({
+      node,
+      messageId: 'emptyRel',
+      fix(fixer) {
+        return fixer.remove(node);
+      }
+    });
+  }
+
+  const parts = splitIntoRangedParts(node);
+  for (const part of parts) {
+    const allowedTags = standardValues.get(part.value);
+    if (!allowedTags) {
+      context.report({
+        node,
+        messageId: 'realRelValues',
+        data: {
+          value: part.reportingValue
+        },
+        fix(fixer) {
+          return fixer.removeRange(part.range);
+        }
+      });
+    } else if (!allowedTags.has(parentNodeName)) {
+      context.report({
+        node,
+        messageId: 'matchingRelValues',
+        data: {
+          value: part.reportingValue,
+          tag: parentNodeName
+        },
+        fix(fixer) {
+          return fixer.removeRange(part.range);
+        }
+      });
+    }
+  }
+}
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    docs: {
+      description: 'Forbid `rel` attribute with an invalid value`',
+      category: 'Possible Errors',
+      url: docsUrl('jsx-no-invalid-rel')
+    },
+    messages: {
+      relOnlyOnSpecific: 'The "rel" attribute only has meaning on `<link>`, `<a>`, `<area>`, and `<form>` tags.',
+      emptyRel: 'An empty "rel" attribute is meaningless.',
+      onlyStrings: '"rel" attribute only supports strings',
+      realRelValues: '{{ value }} is never a valid "rel" attribute value.',
+      matchingRelValues: '"{{ value }}" is not a valid "rel" attribute value for <{{ tag }}>.'
+    }
+  },
+
+  create(context) {
+    return {
+      JSXAttribute(node) {
+        // ignore attributes that aren't "rel"
+        if (node.type !== 'JSXIdentifier' && node.name.name !== 'rel') {
+          return;
+        }
+
+        const parentNodeName = node.parent.name.name;
+        if (!components.has(parentNodeName)) {
+          return context.report({
+            node,
+            messageId: 'relOnlyOnSpecific',
+            fix(fixer) {
+              return fixer.remove(node);
+            }
+          });
+        }
+
+        if (!node.value) {
+          return context.report({
+            node,
+            messageId: 'emptyRel',
+            fix(fixer) {
+              return fixer.remove(node);
+            }
+          });
+        }
+
+        if (node.value.type === 'Literal') {
+          return checkLiteralValueNode(context, node.value, parentNodeName);
+        }
+
```

Comment:
```suggestion
         if (node.value.type !== 'JSXExpressionContainer') {
          return;
        }
```
For readability it is nice to always assert node types.